olive/hf_utils.py

```diff
@@ -0,0 +1,102 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+from typing import Callable
+
+import torch
+
+
+def load_huggingface_model_from_task(task: str, name: str):
+    """Load huggingface model from task and name"""
+    from transformers import AutoModel
+    from transformers.pipelines import check_task
+
+    task_results = check_task(task)
+    assert isinstance(task_results, tuple)
+    if len(task_results) == 2:
+        targeted_task = task_results[0]
+    elif len(task_results) == 3:
+        targeted_task = task_results[1]
+    else:
+        raise ValueError("unsupported transfomers version")
+
+    model_class = {"pt": targeted_task["pt"]}
+    class_tuple = ()
+    class_tuple = class_tuple + model_class.get("pt", (AutoModel,))
+
+    model = None
+    for model_class in class_tuple:
+        try:
+            model = model_class.from_pretrained(name)
+            return model
+        except (OSError, ValueError):
+            continue
+
+    return model
+
+
+class ORTWhisperModel(torch.nn.Module):
+    """ORT implementation of whisper model"""
+
+    def __init__(self, encoder_decoder_init: torch.nn.Module, decoder: torch.nn.Module, config):
+        super().__init__()
+        self.encoder_decoder_init = encoder_decoder_init
+        self.decoder = decoder
+        self.config = config
+
+
+def get_ort_whisper_for_conditional_generation(name: str):
+    """Load ORT implementation of whisper model"""
+    from onnxruntime.transformers.models.whisper.whisper_decoder import WhisperDecoder
+    from onnxruntime.transformers.models.whisper.whisper_encoder_decoder_init import WhisperEncoderDecoderInit
+    from transformers import WhisperForConditionalGeneration
+
+    # get the original model
+    model = WhisperForConditionalGeneration.from_pretrained(name)
+
+    # ort implementation
+    decoder = WhisperDecoder(model, None, model.config)
+    encoder_decoder_init = WhisperEncoderDecoderInit(
+        model,
+        model,
+        None,
+        model.config,
+        decoder_start_token_id=None,
+    )
+
+    return ORTWhisperModel(encoder_decoder_init, decoder, model.config)
+
+
+MODEL_CLASS_TO_ORT_IMPLEMENTATION = {"WhisperForConditionalGeneration": get_ort_whisper_for_conditional_generation}
```

Comment:
I would suggest the key should be task type in huggingface pipeline. see `load_huggingface_model_from_task`.
The workflow should be like:
1. provide `task_name` & `model_name`(hf model name)
2. get model loader from `load_huggingface_model_from_task` like `WhisperForConditionalGeneration` used in line 53.
3. use the model loader to load model.
4. For specific model, you can apply different type function for ort loading with different task types.
Currently, the key for this map containers Model name + task type which will increase the map size a lot in the future.

But if you think the changes will increase the dev costs, please add TODO for future refactoring.

Reply:
Oh, I see your implementation for task load.

Reply:
I was thinking about something similar too. `model_class` doesn't seem like a sustainable way of doing it. Especially when we will probably need the hf_config for dataset creation in the future too. 
I will add a TODO for refactoring since I don't think I can do it in time for this PR which is needed for other work items (beam search, etc)